google/cloud/sql/connector/connector.py

```diff
@@ -17,6 +17,54 @@
 import googleapiclient
 
 
+def get_metadata(service, proj_name, inst_name):
+    """
+    A helper function that requests metadata from the Cloud SQL Instance
+    and returns a dictionary containing the IP addresses and certificate
+    authority of the Cloud SQL Instance.
+
+    Args:
+        service (googleapiclient.discovery.Resource): A service object created
+          from the Google Python API client library. Must be using the SQL
+          Admin API. For more info check out
+          https://github.com/googleapis/google-api-python-client.
+        proj_name (str): A string representing the name of the project.
+        inst_name(str): A string representing the name of the instance.
+            Usually found in environment variable 'CLOUD_SQL_INSTANCE_NAME.'
+
+    Returns:
+        Returns a dictionary containing a dictionary of all IP addresses and
+        their type and a string representing the certificate authority.
+
+    Raises:
+        TypeError: If any of the arguments are not the specified type.
+    """
+
+    if (
+        not isinstance(service, googleapiclient.discovery.Resource)
+        or not isinstance(proj_name, str)
+        or not isinstance(inst_name, str)
+    ):
+        raise TypeError(
+            "Arguments must be as follows: "
+            + "service (googleapiclient.discovery.Resource), "
+            + "proj_name (str) and inst_name (str)."
+        )
+
+    req = service.instances().get(project=proj_name, instance=inst_name)
+    res = req.execute()
+
+    # Extract server certificate authority
+    serverCaCert = res["serverCaCert"]["cert"]
```

Comment:
is there a chance these properties aren't in the result?

Reply:
Don't think so, I'm pretty sure it's required in the response.

Here's the link to the API: https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/instances

Reply:
From previous experience, I'm pretty sure it would result in an earlier error. I'll add in a check just to be sure!